extensions/compose/src/cli-run.spec.ts

```diff
@@ -0,0 +1,62 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { beforeEach, expect, test, vi } from 'vitest';
+import * as extensionApi from '@podman-desktop/api';
+import type { OS } from './os';
+import { CliRun } from './cli-run';
+
+vi.mock('@podman-desktop/api', async () => {
+  return {
+    window: {
+      showInformationMessage: vi.fn().mockReturnValue(Promise.resolve('Yes')),
+      showErrorMessage: vi.fn(),
+      withProgress: vi.fn(),
+      showNotification: vi.fn(),
+    },
+    ProgressLocation: {
+      APP_ICON: 1,
+    },
+  };
+});
+
+beforeEach(() => {
+  vi.clearAllMocks();
+});
+
+test('error: expect installBinaryToSystem to fail with a non existing binary', async () => {
+  // Mock the platform to be linux
+  Object.defineProperty(process, 'platform', {
+    value: 'linux',
+  });
+
+  // Create the CliRun object to call installBinaryToSystem
+  let fakeContext: extensionApi.ExtensionContext;
+  let fakeOs: OS;
+  const cliRun = new CliRun(fakeContext, fakeOs);
+
+  // Run installBinaryToSystem with a non-binary file
+  try {
+    await cliRun.installBinaryToSystem('test', 'tmpBinary');
+    // Expect that showErrorMessage is called
+    expect(extensionApi.window.showErrorMessage).toHaveBeenCalled();
+  } catch (err) {
+    expect(err).to.be.a('Error');
```

Comment:
Why do we have those 2 expect ?

Reply:
To make sure that the message shows it as an actual 'Error'. Sometimes testing gives a false positive when it cant display the message correctly, so I always check the error output.

Reply:
Redid the test! Thanks for the input

Reply:
you can also use toThrow('error message') to verify which error message is thrown (if it's the expected one and not another error)